src/language_features/hover.rs

```diff
@@ -71,17 +72,29 @@ pub fn editor_hover(
                 .join("\n")
         })
         .unwrap_or_else(String::new);
+
     let contents = match result {
         None => "".to_string(),
         Some(result) => match result.contents {
-            HoverContents::Scalar(contents) => contents.plaintext(),
+            HoverContents::Scalar(contents) => {
+                let markdown = match contents {
+                    MarkedString::String(s) => s,
+                    MarkedString::LanguageString(s) => format!("```\n{}\n```", s.value),
```

Comment:
I believe if the payload contains three consecutive backticks, then we need to one-up that by using four surrounding backticks, and so on.
I'm not 100% sure if our markdown parser works that way, so maybe test that.